### ISSUE TITLE
We can remove `failOnError` flag

### ISSUE THREAD

**Issue by volodya-lombrozo** [objectionary/eo#1708]
The `failOnError` flag was added to solve that [issue](https://github.com/objectionary/eo/issues/694) in order to continue build even if some errors is found, but what if we make that behaviour as default? I mean: it seems good to parse and optimize as many files as we can for any build and only when the entire `assemble` phase is finished we can decide what to do with the `errors` and `warnings`.

What I suggest:
1. Assemble all `eo` files without checking `warnings` and `errors`  (you can imagine for now like we have `failOnWarning=false` and `failOnError=false`)
2. Add a `CheckErrorsMojo` phase after the `AssemblyMojo`
2.1 The `CheckErrorsMojo` will check the `failOnWarning` flag and throw an exhaustive exception if we have `warnings`
2.2 The `CheckErrorsMojo` will check the `errors` and throw an exhaustive exception if at least one is found

After implementation:
1. We will be able to remove the `failOnError` flag at all
2. We will always have entire list of warnings and errors for all files
3. We will be able to terminate build if `failOnWarning=true`

In other words, I suggest to check warnings and errors after entire `assembly` phase (not during).

**Comment by yegor256** [comment]
@volodya-lombrozo perfect idea, but let's call it `VerifyMojo`

**Comment by yegor256** [comment]
@Graur please, help here

**Comment by yegor256** [comment]
@Graur what's up with this one?

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo 
> 2.2 The CheckErrorsMojo will check the errors and throw an exhaustive exception if at least one is found
What if we want to skip errors?

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon It could lead to unpredictable results. I don't think we should allow it.

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo even for the test purposes?

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon Even for tests. To be honest, it is hard to imagine the test that will require this. Maybe you have an example?

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo
<img width="641" alt="image" src="https://github.com/objectionary/eo/assets/37025995/09466894-6302-41a5-b310-3a60dd227ec5">

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo almost all of the tests above actually test that "something fails with `failOnError=true`"

**Comment by volodya-lombrozo** [comment]
@maxonfjvipon Maybe. I'm not sure, to be honest. It's better to check that tests

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo what about `fail-on-critical`? Where should we check critical errors? Also to the end?

**Comment by 0pdd** [comment]
@volodya-lombrozo the puzzle [#2443](https://github.com/objectionary/eo/issues/2443) is still not solved.

**Comment by 0pdd** [comment]
@volodya-lombrozo the puzzle [#2678](https://github.com/objectionary/eo/issues/2678) is still not solved; solved: [#2443](https://github.com/objectionary/eo/issues/2443).

**Comment by 0pdd** [comment]
@volodya-lombrozo all 2 puzzles are solved here: [#2443](https://github.com/objectionary/eo/issues/2443), [#2678](https://github.com/objectionary/eo/issues/2678).

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo `VerifyMojo` was introduced and `failOnError` flag was removed [here](https://github.com/objectionary/eo/pull/2694). If everything is ok, please close the issue.

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo since there's no activity here, I'm closing it
